### PR TITLE
feat: implement `Image` object

### PR DESCRIPTION
**Pull request by Dworv**
This PR conveniently implements the `Image` object designed for ease of use with attrs. It features a dynamic converter function that allow for easily providing a path within `cdn.discordapp`.

### CODE REVIEW

**Comment by Dworv** [comment]
This is ready now.

File: retux/client/resources/abc.py [hunk 1]
```diff
@@ -96,6 +97,303 @@ def increment(self) -> int:
...
+
+    def _c(endpoint: CDNEndpoint):
+        """
+        For internal use only.
+
+        Builds an attrs converter for Images.
+        A `__post_attrs_init__` must be used
+        to add extra variables needed in the
+        endpoint such as ids. This is only 
+        meant for endpoints that need a hash.
```
Comment (i0bs) [inline retux/client/resources/abc.py:359]:
```suggestion
        meant for endpoints that need a hash.
        
        Please note that the naming of this method, `_c`
        does not reflect the same serialisation methods
        or strategies as `Serialisable._c()`.
```

File: retux/client/resources/abc.py [hunk 2]
```diff
@@ -96,6 +97,303 @@ def increment(self) -> int:
...
+        Whether or not the image is animated.
+    path : `str`
+        The path to the image in CDN.
+    url : `str`
+        The url to the image.
+    """
+    hash: str | None = field(default=None)
+    """The hash of the image."""
+    _endpoint: CDNEndpoint = field()
+    """Internal use only. The endpoint of the image."""
```
Comment (i0bs) [inline retux/client/resources/abc.py:344]:
```suggestion
    """The endpoint of the image."""
```

File: retux/client/resources/abc.py [hunk 3]
```diff
@@ -96,6 +97,303 @@ def increment(self) -> int:
...
+    _ids: list | None = field(default=None)
+    """
+    Internal use only. The needed variables aside
```
Comment (i0bs) [inline retux/client/resources/abc.py:347]:
```suggestion
    The needed variables aside 
```

**Review by i0bs — CHANGES_REQUESTED** [review]

**Review by i0bs — APPROVED** [review]
LGTM, thanks for adding the clarification on the method.

**Pull request merged**